generator.cpp

```diff
@@ -13,17 +13,17 @@ std::unordered_set<uint64_t> visited;
  * Explore and print all possible position under a given depth.
  * symetric positions are printed only once.
  */
-void explore(const Position &P, char* pos_str, const int depth) {
+void explore(const Position &P, char* const pos_str, const int depth) {
```

Comment:
Added a const-annotation to the pointer here to ensure memory safety and optimization

Reply:
This parameter is actually not const.
The code does not even compile with your suggestion.

Reply:
Wait, the integer values are variable, but the pointer is `const`, no? I compiled on my machine and all of this worked;.

Reply:
my bad, sorry I misplaced the const when copied the code.